.github/workflows/upload-pypi.yaml

```
@@ -0,0 +1,51 @@
+# Copyright 2021 The GPflux Contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Upload-PyPi
+
+on:
+  workflow_dispatch:
+  push:
+    tags: v[0-9]+.[0-9]+.[0-9]+*
```

is that `+*` at the end correct?

---

I believe so, can be used to publish release candidates (e.g., v0.2-rc1).

---

should that then not be `.*`?
```suggestion
    tags: v[0-9]+.[0-9]+.[0-9]+.*
```

---

assuming this is regexp, actually we probably want to escape the literal `.`:
```suggestion
    tags: v[0-9]+\.[0-9]+\.[0-9]+.*
```
?

---

I ignorantly just copied this from the Bellman repo, but I will double check that this does the right thing.

---

Looks like github actions does *not* use regex for the tags and the original pattern was good as is. See: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

---

thanks for checking!